### DISCUSSION TITLE
time format

### DISCUSSION THREAD

**Discussion by Deba57** [dtcenter/METplus#2190] [Configuration] [answered]
Hi,
I am using PDY format (for example PDY=20230223) for valid date, how do I translate that to %Y%m%d_%H%M%S format,as observation I am reading are in %Y%m%d_%H%M%S . 
I am getting ERROR: Could not format time string (20230223) using time format (%Y%m%d_%H%M%S)

Thanks
Debasish.

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hi @Deba57, I'm not familiar with the "PDY" date format. Couldn't find any mention of it for the unix date command or for python. 

But I did eventually find it in this [WCOSS Implementation Standards](https://www.nco.ncep.noaa.gov/idsb/implementation_standards/ImplementationStandards.v11.0.0.pdf) document, where:
`PDY = Date in YYYYMMDD format`. So PDY is a J-job variable.

I'm not exactly sure of the context in which you're getting this runtime error. But I'm guessing that you're running METplus on an HPC where PDY is defined as an environment variable and you'd like to reference its value within the METplus configuration file.

The [METplus User's Guide](https://metplus.readthedocs.io/en/latest/Users_Guide/systemconfiguration.html#setting-config-variables-with-environment-variables) mentions how you can reference the value of environment variables.

If you wanted to reference the PDY to set the VALID_BEG settings in METplus, you could try this:
```
VALID_TIME_FMT = %Y%m%d
VALID_BEG = {ENV[PDY]}
```

Of course, I haven't actually tested this so I'm not sure. If that doesn't get you past this issue, please send me more information about where you're trying to reference PDY, along with a sample METplus config file.

**Reply by JohnHalleyGotway** [reply]
@Deba57 just wanted to followup on this discussion. Are you still stuck or were you able to figure it out?

**Comment by Deba57** [comment]
Thanks @JohnHalleyGotway . It is sorted now.
